common/softwarecontainer-common.cpp

```diff
@@ -192,14 +196,42 @@ ReturnCode FileToolkitWithUndo::createDirectory(const std::string &path)
     return ReturnCode::SUCCESS;
 }
 
-ReturnCode FileToolkitWithUndo::bindMount(const std::string &src, const std::string &dst, bool readOnly)
+std::string FileToolkitWithUndo::tempDir(std::string templ)
+{
+    char *dir = const_cast<char*>(templ.c_str());
+    dir = mkdtemp(dir);
+    if (dir == NULL) {
+        log_warning() << "Failed to create buffered Directory: " << strerror(errno);
+        return nullptr;
+    }
+
+    return StringBuilder() << dir;
+}
```

Comment:
Is this tmpdir cleaned up on exit, like the other things in this class? If not - it should be!